logisland-components/logisland-processors/logisland-processor-web-analytics/src/main/java/com/hurence/logisland/processor/webAnalytics/URLDecoder.java

```diff
@@ -104,20 +114,35 @@ private void updateRecord(ProcessContext context, Record record, HashSet<String>
         }
         fields.forEach(fieldName -> {
             // field is already here
-            if (record.hasField(fieldName))  {
+            if (record.hasField(fieldName)) {
                 String value = record.getField(fieldName).asString();
-                try {
-                    String decodedValue = java.net.URLDecoder.decode(value, charset);
-                    if (! decodedValue.equals(value)) {
-                        final FieldType fieldType = record.getField(fieldName).getType();
-                        record.removeField(fieldName);
-                        record.setField(fieldName, fieldType, decodedValue);
-                    }
-                } catch (UnsupportedEncodingException e) {
-                    logger.warn(e.toString());
-                }
+                decode(value,charset,record,fieldName,true);
             }
         });
     }
 
+    private void decode (String value, String charset, Record record, String fieldName, boolean tryTrick)
+    {
+        String decodedValue = null;
+        try {
+            decodedValue = java.net.URLDecoder.decode(value, charset);
+            if (!decodedValue.equals(value)) {
+                final FieldType fieldType = record.getField(fieldName).getType();
+                record.removeField(fieldName);
+                record.setField(fieldName, fieldType, decodedValue);
+            }
+        } catch (IllegalArgumentException e) {
+            if (tryTrick) {
+                value = value.replaceAll("%(?![0-9A-F]{2})", percentEncodedChar);
```

Comment:
the regex should be "%(?![0-9a-fA-F]{2})" to be case insensitive.
```suggestion
                value = value.replaceAll("%(?![0-9a-fA-F]{2})", percentEncodedChar);
```

Reply:
I agree